src/common.rs

```diff
@@ -31,9 +31,9 @@ pub struct Hash256([u8; 32]);
 impl Serialize for Hash256 {
     fn serialize<S: serde::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
         if serializer.is_human_readable() {
-            serializer.serialize_str(&self.to_string())
+            String::serialize(&self.to_string(), serializer)
```

Comment:
FYI: these renames are for consistency rather than functionality. In theory `&self.to_string().serialize(&serializer)` might also work but specifying the intended type first and then calling `serialize` on it seemed like the most readable option to me.